packages/utils/src/dsn.ts

```diff
@@ -3,7 +3,7 @@ import { DsnComponents, DsnLike, DsnProtocol } from '@sentry/types';
 import { SentryError } from './error';
 
 /** Regular expression used to parse a Dsn. */
-const DSN_REGEX = /^(?:(\w+):)\/\/(?:(\w+)(?::(\w+))?@)([\w.-]+)(?::(\d+))?\/(.+)/;
+const DSN_REGEX = /^(?:(\w+):)\/\/(?:(\w+)(?::(\w+)*)?@)([\w.-]+)(?::(\d+))?\/(.+)/;
```

Comment:
I might be missing something in terms of our DSN spec but considering that the added `*` seems to have ReDoS potential (as pointed out by the bot warning), we might want to change the `*` to `?`. 

Haven't tried it out in the SDK code itself but after a quick check on [regex101](https://regex101.com/r/RVSGmx/1), it seems like this still works. 
 
Changing `*` to `?` probably gets rid of the ReDoS warning (checked with an [online ReDoS checked](https://devina.io/redos-checker))

WDYT, does that sound reasonable?

```suggestion
const DSN_REGEX = /^(?:(\w+):)\/\/(?:(\w+)(?::(\w+)?)?@)([\w.-]+)(?::(\d+))?\/(.+)/;
```

Reply:
Oh yeah you're right. `?` should be enough in this case. 👍🏻